src/Func.cpp

```diff
@@ -1704,6 +1704,37 @@ Stage &Stage::gpu_tile(VarOrRVar x, VarOrRVar y, VarOrRVar z,
     return gpu_tile(x, y, z, bx, by, bz, tx, ty, tz, x_size, y_size, z_size, tail, device_api);
 }
 
+Stage &Stage::shader(VarOrRVar x, VarOrRVar y, VarOrRVar c, DeviceAPI device_api) {
+    reorder(c, x, y);
+    // GLSL outputs must be stored interleaved
+    // No reorder_storage, so we manually reorder
+    vector<StorageDim> reordered_storage;
```

Comment:
This seems fishy. I don't think reordering storage like this does anything if this isn't the pure definition. That's why the reorder_storage API doesn't exist on Stage.

Maybe Func::shader should do the reordering, but Stage::shader should not?